src/Test.FunctionalTests.Sarif/TestData/Multitool/ValidateCommand/Inputs/SARIF2004.OptimizeFileSize_Valid.sarif

```diff
@@ -42,6 +42,25 @@
               }
             }
           ]
+        },
+        {
+          "ruleIndex": -1,
+          "rule": {
+            "id": "TEST1001",
+            "toolComponent": { "name": "Plugin01" }
+          },
+          "message": { "text": "Good result: artifacts array provides additional useful information." }
+        },
+        {
+          "ruleId": "TEST1001",
+          "ruleIndex": 0,
+          "locations": [
+            {
+              "physicalLocation": { "artifactLocation": { "uri": "local/test/code/src/file1.h" } }
+            }
+          ],
+          "analysisTarget": { "uri": "local/test/code/src/file1.c" },
+          "message": { "text": "Good result: artifacts array provides additional useful information." }
```

Comment:
>artifacts array provides additional useful information [](start = 45, length = 54)

The `artifacts` array doesn't contain additional information. All it has is a `location`. (I understand that Line 62 doesn't have `uriBaseId`, and the `artifacts` array does, but still, the intent of the rule is that the `artifacts` array should provide something besides location information. In any case, why are we adding this test? Don't we already have tests for non-useful artifacts arrays?

Reply:
Besides the point that i added a non-absoluteuri, this testing is a good one saying that we have a location.physicallocation.artifactlocation.uri != analysisTarget. The message is wrong and i will solve it

---
In reply to: [454706843](https://github.com/microsoft/sarif-sdk/pull/1995#discussion_r454706843) [](ancestors = 454706843)